view-generator-framework/src/main/java/org/hypertrace/core/viewgenerator/service/ViewGenerationProcessor.java

```diff
@@ -48,15 +43,14 @@ public void init(ProcessorContext context) {
   }
 
   @Override
-  public List<KeyValue<String, OUT>> transform(String key, IN value) {
-    List<KeyValue<String, OUT>> outputKVPairs = new ArrayList<>();
-    List<OUT> output = viewGenerator.process(value);
+  public void process(Record<String, IN> record) {
+    long nowMillis = Instant.now().toEpochMilli();
+    List<OUT> output = viewGenerator.process(record.value());
     if (output != null) {
       for (OUT out : output) {
-        outputKVPairs.add(KeyValue.pair(key, out));
+        context.forward(new Record<>(record.key(), out, nowMillis));
```

Comment:
what's the millis input on the record represent?

Reply:
record generation timestamp. will be useful later for the calculation of latencies.

<img width="1052" alt="image" src="https://github.com/hypertrace/view-generator-framework/assets/60599147/54529360-d81f-4feb-a65c-04d979e493fb">

Reply:
So the output record - shouldn't its timestamp be the time it's forwarded (i.e. after we call the process function)? Could be I'm misunderstanding here though.

Reply:
> shouldn't its timestamp be the time it's forwarded

Yes. Will move the ts post process.

To be very precise and perfect, we can even generate a timestamp for output record.
But, I was just trying to save some unnecessary gc here by not creating a timestamp in a loop.